packages/page-parachains/src/Teleport.tsx

```diff
@@ -77,7 +77,7 @@ function Teleport ({ onClose }: Props): React.ReactElement<Props> | null {
   const params = useMemo(
     () => [
       {
-        V1: isParaTeleport
+        [xcmVersion]: isParaTeleport
```

Comment:
This really is going to be problematic down the line - we are assuming here that all future versions will have the same "form", just because current versions have that form. So it will break at some point.

(We already had the V0 -> V1 swap which was significantly different and certainly not compatible)

I would suggest just supporting the current V2 here - if some chains have not been enabled, they just need to be disabled teleport-wise.

Reply:
Thank you for the comment. Yes, this might break with introduction of XCM v3.

The teleport between Polkadot and Statemint is via XCM v1. When trying with XCM v2 I received an error similar to the one shown [here](https://github.com/polkadot-js/apps/issues/9127#issue-1615301770).

If we only allow V2, the most important teleport, the one between Polkadot and Statemint, will not be supported.

I could add a comment and/or try to restrict the type of xcmVersion to "V1" and "V2" in typescript.

Reply:
How to proceed?

Reply:
@jacogr I restricted the xcm version via string literal types to V2 and V1. Therefore the build would break if anyone tries using v3. Also I added a comment.

Could you have look?